packages/blocks/src/api/parser/convert-legacy-block.js

```diff
@@ -77,6 +77,35 @@ export function convertLegacyBlockNameAndAttributes( name, attributes ) {
 		newAttributes.legacy = true;
 	}
 
+	// Grid layout attributes were stored as strings in WP 6.6. Convert them to numbers.
```

Comment:
Apparently `columnCount` shipped in 6.3. Need to check that though.

```suggestion
	// Grid layout attributes were stored as strings in WP 6.3-6.6. Convert them to numbers.
```

Reply:
It did! See [here](https://make.wordpress.org/core/2023/07/14/layout-updates-in-the-editor-for-wordpress-6-3/) for details. The manual mode (using `columnCount`) was added to Post Template too, so it was definitely in use from that point onwards. The good news is in TT4 markup (see [here](https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-content/themes/twentytwentyfour/patterns/posts-images-only-3-col.php#L20) for instance) it's always a number.